pkg/runner/run_context.go

```diff
@@ -244,57 +244,38 @@ func (rc *RunContext) interpolateOutputs() common.Executor {
 	}
 }
 
-// Executor returns a pipeline executor for all the steps in the job
-func (rc *RunContext) Executor() common.Executor {
-	steps := make([]common.Executor, 0)
-
-	steps = append(steps, func(ctx context.Context) error {
-		if len(rc.Matrix) > 0 {
-			common.Logger(ctx).Infof("\U0001F9EA  Matrix: %v", rc.Matrix)
-		}
-		return nil
-	})
+func (rc *RunContext) startContainer() common.Executor {
+	return rc.startJobContainer()
+}
 
-	steps = append(steps, rc.startJobContainer())
+func (rc *RunContext) stopContainer() common.Executor {
+	return rc.stopJobContainer()
+}
 
-	for i, step := range rc.Run.Job().Steps {
-		if step.ID == "" {
-			step.ID = fmt.Sprintf("%d", i)
+func (rc *RunContext) closeContainer() common.Executor {
+	return func(ctx context.Context) error {
+		if rc.JobContainer != nil {
+			return rc.JobContainer.Close()(ctx)
 		}
-		stepExec := rc.newStepExecutor(step)
-		steps = append(steps, func(ctx context.Context) error {
-			err := stepExec(ctx)
-			if err != nil {
-				common.Logger(ctx).Errorf("%v", err)
-				common.SetJobError(ctx, err)
-			} else if ctx.Err() != nil {
-				common.Logger(ctx).Errorf("%v", ctx.Err())
-				common.SetJobError(ctx, ctx.Err())
-			}
-			return nil
-		})
+		return nil
 	}
-	steps = append(steps, func(ctx context.Context) error {
-		err := rc.stopJobContainer()(ctx)
-		if err != nil {
-			return err
-		}
+}
 
-		rc.Run.Job().Result = "success"
-		jobError := common.JobError(ctx)
-		if jobError != nil {
-			rc.Run.Job().Result = "failure"
-		}
+func (rc *RunContext) matrix() map[string]interface{} {
+	return rc.Matrix
+}
 
-		return nil
-	})
+func (rc *RunContext) result(result string) {
+	rc.Run.Job().Result = result
+}
 
-	return common.NewPipelineExecutor(steps...).Finally(rc.interpolateOutputs()).Finally(func(ctx context.Context) error {
```

Comment:
This issue #758 was closed on November 2021, and from the comments it could be understood that the issue was actually fixed here #894. It seems like the changes introduced in this file might have caused the issue to respawn. I reproduced it in my local environment by creating a simple job with two steps, and it was only possible to set output variables in the first one.
@Poltergeist I would really appreciate it if you could verify that the fix from November 2021 was affected by these changes.

Reply:
Hi @cweckesser, can you create a new issue with a reproducible workflow to show this case?